pkg/plugin/kms_v2_server_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft and contributors.  All rights reserved.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/Azure/kubernetes-kms/pkg/metrics"
+	mockkeyvault "github.com/Azure/kubernetes-kms/pkg/plugin/mock_keyvault"
+
+	"github.com/Azure/kubernetes-kms/pkg/version"
+	kmsv2 "k8s.io/kms/apis/v2"
+)
+
+func TestV2Encrypt(t *testing.T) {
+	tests := []struct {
+		desc   string
+		input  []byte
+		output []byte
+		err    error
+	}{
+		{
+			desc:   "failed to encrypt",
+			input:  []byte("foo"),
+			output: []byte{},
+			err:    fmt.Errorf("failed to encrypt"),
+		},
+		{
+			desc:   "successfully encrypted",
+			input:  []byte("foo"),
+			output: []byte("bar"),
+			err:    nil,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			kvClient := &mockkeyvault.KeyVaultClient{}
+			kvClient.SetEncryptResponse(test.output, test.err)
+
+			kmsV2Server := KeyManagementServiceV2Server{
+				kvClient: kvClient,
+				reporter: metrics.NewStatsReporter(),
+			}
+
+			out, err := kmsV2Server.Encrypt(context.TODO(), &kmsv2.EncryptRequest{
+				Plaintext: test.input,
+			})
+			if err != test.err {
+				t.Fatalf("expected err: %v, got: %v", test.err, err)
+			}
+			if string(out.GetCiphertext()) != string(test.output) {
```

Comment:
```suggestion
			if !bytes.Equal(out.GetCiphertext(), test.output) {
```

applies in other places too